File: problem006/6.lua
#!/usr/bin/env lua

function sum_of_squares(n)
	return (n * (n + 1) * ((2 * n) + 1)) / 6
end

function square_of_sum(n)
	if n % 2 == 0 then
		return ((n / 2) * (n + 1)) ^ 2
	else
		return (((n / 2) * (n + 1)) + n) ^ 2
	end
end

print(square_of_sum(100) - sum_of_squares(100))

File: problem006/6.pl
#!/usr/bin/env perl
use strict;
use warnings;

sub sum_of_squares{
	my $n = shift;
	return ($n * ($n + 1) * ((2 * $n) + 1)) / 6;
}

sub square_of_sum{
	my $n = shift;
	my $sum = ($n / 2) * ($n + 1);
	if ($n % 2 != 0) { $sum += $n; }
	return $sum ** 2;
}

print(square_of_sum(100) - sum_of_squares(100));
